alws/crud.py

```diff
@@ -181,7 +181,44 @@ async def get_available_build_task(
         if not db_task:
             return
         db_task.ts = datetime.datetime.now()
-        db_task.status = BuildTaskStatus.STARTED
+        db_task.status = BuildTaskStatus.BUILD_STARTED
+        await db.commit()
+    return db_task
+
+
+async def get_available_sign_task(
+            db: Session, 
+            request: build_task_schema.RequestTask
+        ) -> models.BuildTask:
+    async with db.begin():
+        ts_expired = datetime.datetime.now() - datetime.timedelta(minutes=20)
+        query = ~models.BuildTask.dependencies.any()
+        db_task = await db.execute(
+            select(models.BuildTask).where(query).with_for_update().filter(
+                    sqlalchemy.and_(
+                        models.BuildTask.status < BuildTaskStatus.SIGN_COMPLETED,
+                        models.BuildTask.status >= BuildTaskStatus.BUILD_COMPLETED,
+                        models.BuildTask.arch.in_(request.supported_arches),
```

Comment:
rpmsign doesn't require from rpm to be same arch as host machine, so, `supported_arches` can be removed